lib/gameboard/board.rb

```diff
@@ -0,0 +1,495 @@
+    end
+
+    # Public: Check if the gameboard is full relative to the default cell.
+    #
+    #
+    #
+    # Example
+    #
+    #    Board.new(height: 3, width: 3)
+    #      -------------
+    #    0 | 1 | 2 | 3 |
+    #      -------------
+    #    1 | A | B | C |
+    #      -------------
+    #    2 | X | Y | Z |
+    #      -------------
+    #      - 0 - 1 - 2 -
+    #
+    #   full?
+    #     #=> true
+    #
+    #   Board.new(height: 3, width: 3, cells: "Y")
+    #
+    #   full?
+    #     #=> false (because our default piece is now "Y")
+    #
+    def full?
+      board.none? { |cell| cell.value == cells  }
+    end
+
+    # Public: Return every row on the gameboard.
+    #
+    # coords: An optional boolean to specify returning an Array of coordinates
+    #         instead of values.
+    #
+    # Example
+    #      -------------
+    #    0 | 1 | 2 | 3 |
+    #      -------------
+    #    1 | A | B | C |
+    #      -------------
+    #    2 | X | Y | Z |
+    #      -------------
+    #      - 0 - 1 - 2 -
+    #
+    #
+    #   horizontal
+    #     #=> [
+    #          ["X", "Y", "Z"],
+    #          ["A", "B", "C"],
+    #          [1, 2, 3]
+    #         ]
+    #
+    #   horizontal(true)
+    #     #=> [
+    #          [[0, 0], [1, 0], [2, 0]],
+    #          [[0, 1], [1, 1], [2, 1]],
+    #          [[0, 2], [1, 2], [2, 2]]]
+    #         ]
+    #
+    def horizontal(coords = false)
+      rows = []
+      height.times do |y|
+        rows << board.select { |cell| cell.coord.y == y }.map do |cell|
+          coords ? cell.coord.position : cell.value
+        end
+      end
+      rows
+    end
+
+    # Public: Return all valid neighbors of a given coordinate.
+    #
+    # point  - The X,Y coordinate neighbors should be based off of.
+    # coords - An optional boolean to return an array of coordinate pairs
+    #
+    #
+    # Examples
+    #                   -------------
+    #                 0 | 1 | 2 | 3 |
+    #                   -------------
+    #                 1 | A | B | C |
+    #                   -------------
+    #                 2 | X | Y | Z |
+    #                   -------------
+    #                   - 0 - 1 - 2 -
+    #
+    #   neighbors([1,1])
+    #           #=> [1, 2, 3, "A", "C", "X", "A", 1]
+    #
+    #   neighbors([1,1] true)
+    #           #=> [
+    #                [0, 2], [1, 2],
+    #                [2, 2], [0, 1],
+    #                [2, 1], [0, 0],
+    #                [0, 1], [0, 2]
+    #               ]
+    #
+    #
+    def neighbors(point, coords = false)
+      temp = Coordinate.new(point[0], point[1])
+      valid_neighbors(temp).map do |cell|
+        coords ? cell.coord.position : cell.value
+      end
+    end
+
+    # Public: Set random cells to a given value.
+    #
+    # piece - A value to set to cells
+    #
+    def randomize(piece)
+      rand(1...board.length).times do
+        board[rand(board.length)].value = piece
```

Comment:
This method's a good idea. Maybe the min/max # of repetitions could be optionally passed in.

Random_cell could also be a nice public method that this method could use in its implementation.